tests/test_vonkarman.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) 2012-2017 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+from __future__ import print_function
+import numpy as np
+import os
+import sys
+
+from galsim_test_helpers import *
+
+try:
+    import galsim
+except ImportError:
+    sys.path.append(os.path.abspath(os.path.join(path, "..")))
+    import galsim
+
+
+@timer
+def test_vk(slow=False):
+    """Test the generation of VonKarman profiles
+    """
+    if __name__ == '__main__' and slow:
+        lams = [300.0, 500.0, 1100.0]
+        r0_500s = [0.05, 0.15, 0.3]
+        L0s = [1e10, 25.0, 10.0]
+        do_deltas = [False, True]
+    else:
+        lams = [500.0]
+        r0_500s = [0.2]
+        L0s = [25.0]
+        do_deltas = [False]
+    for lam in lams:
+        for r0_500 in r0_500s:
+            r0 = r0_500*(lam/500)**(6./5)
+            for L0 in L0s:
+                for do_delta in do_deltas:
+                    kwargs = {'lam':lam, 'r0':r0, 'L0':L0, 'do_delta':do_delta}
+                    print(kwargs)
+
+                    vk = galsim.VonKarman(flux=2.2, **kwargs)
+                    np.testing.assert_almost_equal(vk.flux, 2.2)
+
+                    check_basic(vk, "VonKarman")
+                    do_pickle(vk)
+                    do_pickle(vk._sbp)
+                    do_pickle(vk._sbp, lambda x: (x.getFlux(), x.getGSParams()))
+
+                    img = galsim.Image(16, 16, scale=0.2)
+                    do_shoot(vk, img, "VonKarman")
+                    do_kvalue(vk, img, "VonKarman")
+
+
+@timer
+def test_vk_delta():
+    """Test a VonKarman with a significant delta-function amplitude"""
+    kwargs = {'lam':1100.0, 'r0':0.8, 'L0':5.0, 'flux':2.2}
+    # Try to see if we can catch the warning first
+    with assert_warns(UserWarning):
+        vk = galsim.VonKarman(**kwargs)
+
+    kwargs['suppress_warning'] = True
+    vk = galsim.VonKarman(**kwargs)
+    do_pickle(vk)
+
+    # This profile has more than 15% of its flux in the delta-function component.
+    np.testing.assert_array_less(0.15, vk.delta_amplitude/vk.flux)
```

Comment:
This would probably read better to just use the assert statement, rather than np.testing.assert_array_less.  (Not least because it's not an array on the rhs.)
```
assert vk.delta_amplitude > 0.15 * vk.flux
```